Tests/L0/SonarQubePreBuild/CreateCommandLineArgs.ps1

```diff
@@ -0,0 +1,67 @@
+[CmdletBinding()]
+param()
+
+. $PSScriptRoot\..\..\lib\Initialize-Test.ps1
+. $PSScriptRoot\..\..\..\Tasks\\SonarQubePreBuild\SonarQubeHelper.ps1
+. $PSScriptRoot\..\..\..\Tasks\\SonarQubePreBuild\SonarQubePreBuildImpl.ps1
+
+
+## Test Case 1 - Minimum configuration 
+
+# Act
+$actual = CreateCommandLineArgs `
+                -projectKey "pkey" `
+                -projectName "Test Project" `
+                -projectVersion "1.0" `
+                -serverUrl "http://localhost:9000" `                
+               
+
+# Assert.
+Assert-AreEqual 'begin /k:"pkey" /n:"Test Project" /v:"1.0" /d:sonar.host.url="http://localhost:9000"' $actual
+
+
+## Test Case 2 - Full configuration 
+
+# Arrange
+$dummyConfigFile = [System.IO.Path]::Combine($PSScriptRoot, "analysis-config.xml");
+New-Item $dummyConfigFile -ItemType File -Force 
+$oldSourcesValue = $env:BUILD_SOURCESDIRECTORY #should be empty
```

Comment:
If it's important to the that the sources directory is empty then the test should check (and if it isn't important then the comment is unnecessary).